interfaces/interfaces.c

```diff
@@ -29,35 +29,32 @@ static const char *entity_list_version = "VClientEntityList003";
 static const char *surface_version = "VGUI_Surface030";
 static const char *vgui_panel_version = "VGUI_Panel009";
 
+/// This is a "bad" way to get the dl paths. Since the dynamic library has all the context of the tf2 executable file,
+/// we can just do the paths with in relativity to the executable file, which is much more agnostic to the operating system.
+
 // Arch consts
 // static const char arch_tfbin_path[256] = "/home/pat/.local/share/Steam/steamapps/common/Team Fortress 2/tf/bin/linux64/";
 // static const char arch_bin_path[256] = "/home/pat/.local/share/Steam/steamapps/common/Team Fortress 2/bin/linux64/";
 
 // Debian consts
-static const char ubuntu_tfbin_path[256] = "/home/pat/.local/share/Steam/steamapps/common/Team Fortress 2/tf/bin/linux64/";
-static const char ubuntu_bin_path[256] = "/home/pat/.local/share/Steam/steamapps/common/Team Fortress 2/bin/linux64/";
+// static const char ubuntu_tfbin_paths[256] = "/home/drcoomer/.local/share/Steam/steamapps/common/Team Fortress 2/tf/bin/linux64/";
+// static const char ubuntu_bin_path[256] = "/home/drcoomer/.local/share/Steam/steamapps/common/Team Fortress 2/bin/linux64/";
 
-CreateInterfaceFn get_factory(const char base_path[256], char *lib_name)
-{
-    char path[256];
-    strcpy(path, base_path);
-    strcat(path, lib_name);
+// how I'd like to implement file paths
+#define tfbin_path "./tf/bin/linux64/"
+#define bin_path   "./bin/linux64/"
```

Comment:
I like the concatenation of strings at compile time but I prefer the type safety of `const char *`. I've added back the runtime concatenation and `base_path` and `lib_name` args. This won't have a performance impact since this only runs at initialization.